tests/testthat/test-run.R

```diff
@@ -261,21 +261,54 @@ test_that("can run manually with shared resources", {
 })
 
 
+test_that("can run with shared resources using shorthand arguments", {
+  path <- test_prepare_orderly_example("shared-shorthand")
+  envir <- new.env()
+  id <- orderly_run_quietly("shared-shorthand", root = path, envir = envir)
+  expect_setequal(
+    dir(file.path(path, "archive", "shared-shorthand", id)),
+    c("data.csv", "mygraph.png", "shared-shorthand.R"))
+  meta <- orderly_metadata(id, root = path)
+  expect_equal(nrow(meta$custom$orderly$shared), 1)
+  expect_equal(meta$custom$orderly$shared,
+               data_frame(here = "data.csv", there = "data.csv"))
+  expect_equal(
+    meta$custom$orderly$role,
+    data_frame(path = c("shared-shorthand.R", "data.csv"),
+               role = c("orderly", "shared")))
+})
+
 test_that("can validate shared resource arguments", {
   expect_error(
-    validate_shared_resource(list(), NULL),
+    orderly_shared_resource(),
     "'orderly_shared_resource' requires at least one argument")
+
   expect_error(
-    validate_shared_resource(list(input = c("a", "b")), NULL),
-    "Invalid shared resource 'input': entries must be strings")
+    orderly_shared_resource(c("a", "b")),
+    "All elements of 'arguments to 'orderly_shared_resource'' must be strings")
   expect_error(
-    validate_shared_resource(list(a = 1, b = TRUE, c = "str"), NULL),
-    "Invalid shared resource 'a', 'b': entries must be strings")
-  expect_equal(
-    validate_shared_resource(list(a = "A", b = "B"), NULL),
-    c(a = "A", b = "B"))
-})
+    orderly_shared_resource(a = 1, b = TRUE, c = "str"),
+    "All elements of 'arguments to 'orderly_shared_resource'' must be strings")
+  expect_error(
+    orderly_shared_resource(1, TRUE, "str"),
+    "All elements of 'arguments to 'orderly_shared_resource'' must be strings")
+  expect_error(
+    orderly_shared_resource(a = 1, TRUE, "str"),
+    "All elements of 'arguments to 'orderly_shared_resource'' must be strings")
```

Comment:
These messages are pretty terrible, but I couldn't find a way to deparse the `...` reliably so I've just used a placeholder instead.

I'd almost be tempted to change `orderly_shared_resource` to accept a single files argument instead, making it consistent with all the other functions, including `orderly_dependency`, `orderly_copy_files` and `orderly_resource`, but of course that's a breaking change.